gwpy/cli/cliproduct.py

```diff
@@ -801,13 +803,14 @@ def annotate_save_plot(self, args):
         all_units = set()
         for ts in self.timeseries:
             un = str(ts.unit)
-            if (un != 'undef') & (un != ''):
+            if (un != 'undef') & (un != '') & (un != '*'):
                 all_units.add(un)
             else:
+                ts.override_unit('')
                 all_units.add('Counts')
 
         if len(all_units) == 1:
-            self.units = all_units.pop()
+            self.units = label_to_latex(all_units.pop())
         else:
             self.units = 'Counts'
```

Comment:
@areeda, since 'Counts' is an actual physical unit, I don't think we should use it as the default. Perhaps something like

```python
        if len(all_units) == 1:
            self.units = label_to_latex(all_units.pop())
        elif len(all_units) > 1:
            self.units = 'multiple units'
        else:
            self.units = 'undef'
```